src/app/clusters/scenes-server/SceneHandlerImpl.cpp

```diff
@@ -16,10 +16,138 @@
  */
 
 #include <app/clusters/scenes-server/SceneHandlerImpl.h>
+#include <app/util/endpoint-config-api.h>
 
 namespace chip {
 namespace scenes {
 
+namespace {
+
+using ConcreteAttributePath  = app::ConcreteAttributePath;
+using AttributeValuePairType = app::Clusters::ScenesManagement::Structs::AttributeValuePair::Type;
+
+/// IsAttributeTypeValid
+/// @brief Check if the attribute type is valid for a scene extension field value according to the spec.
+/// @param type Type of the attribute's metadata
+/// @return
+bool IsSceneValidAttributeType(EmberAfAttributeType type)
+{
+    return (type == ZCL_INT8U_ATTRIBUTE_TYPE || type == ZCL_INT16U_ATTRIBUTE_TYPE || type == ZCL_INT32U_ATTRIBUTE_TYPE ||
+            type == ZCL_INT64U_ATTRIBUTE_TYPE || type == ZCL_INT8S_ATTRIBUTE_TYPE || type == ZCL_INT16S_ATTRIBUTE_TYPE ||
+            type == ZCL_INT32S_ATTRIBUTE_TYPE || type == ZCL_INT64S_ATTRIBUTE_TYPE);
+}
+
+/// IsSignedAttribute
+/// @brief Compare the attribute type to the signed attribute types in ZCL and return true if it is signed.
+/// @param attributeType metadata's attribute type
+/// @return
+bool IsSignedAttribute(EmberAfAttributeType attributeType)
```

Comment:
This is re-inventing emberAfIsTypeSigned.  Which I know, is not really exposed as API right now, but if we need it, we should expose it, no?  Possibly with a better name like IsSignedIntegerAttribute.  Or even an IsSignedIntegerAttribute() member function on EmberAfAttributeMetadata.